retrofit2-clients/src/main/java/com/palantir/remoting2/retrofit2/MultiServerRetryInterceptor.java

```diff
@@ -100,16 +100,26 @@ private Request redirectRequest(Request request, String redirectToUri) {
         if (matchingUri == null) {
             throw new IllegalStateException(String.format("Unrecognized server URI in the request %s. "
                             + "Supported servers are %s. Did you use different server URI for the initial request?",
-                    requestUri, uris));
+                    requestUrl, uris));
         }
 
         // ... replace it with the URI of the server to redirect to.
-        String newRequestUrl = requestUri.replaceFirst(matchingUri, redirectToUri);
+        String newRequestUrl = requestUrl.toString().replaceFirst(matchingUri, redirectToUri);
         return request.newBuilder()
                 .url(HttpUrl.parse(newRequestUrl))
                 // Request.this.tag field by default points to request itself if it was not set in RequestBuilder.
                 // We don't want to reference old request in new one - that is why we need to reset tag to null.
                 .tag(request.tag().equals(request) ? null : request.tag())
                 .build();
     }
+
+    private boolean doServersMatch(HttpUrl serverUri, HttpUrl server) {
```

Comment:
- docs, please
- parameters should have symmetric names; for instance `left`, `right`